provider/win_driver_installer_test.py

```diff
@@ -197,3 +200,64 @@ def check_gagent_version(session, test, gagent_pkg_info_cmd,
     if actual_gagent_version != expected_gagent_version:
         test.fail("gagent version is not right, expected is %s but got %s"
                   % (expected_gagent_version, actual_gagent_version))
+
+
+@error_context.context_aware
+def get_drive_letter(session, test, img_size):
+    """
+    Get drive letter.
+
+    :param session: The guest session.
+    :param img_size: image size.
+    """
+    error_context.context("Format data disk", test.log.info)
+    disk_index = utils_disk.get_windows_disks_index(session, img_size)
+    if not disk_index:
+        test.error("Failed to get the disk index of size %s" % img_size)
+    if not utils_disk.update_windows_disk_attributes(session, disk_index):
+        test.error("Failed to enable data disk %s" % disk_index)
+    drive_letter_list = utils_disk.configure_empty_windows_disk(
+        session, disk_index[0], img_size)
+    if not drive_letter_list:
+        test.error("Failed to format the data disk")
+    return drive_letter_list[0]
+
+
+@error_context.context_aware
+def rng_test(session, test, params, vm):
```

Comment:
I will suggest to remove `session` and keep `vm`, then `session = vm.wait_for_login()`, as I suppose to open a session and then close it inside a function would be safer that pass it everywhere.
Or at least `session` is not proper to be listed as a common parameter to `win_installer_test()`, we better not assume that all test need an opened session.

Reply:
In order to use the same interface: '
eval("win_driver_installer_test.%s_test( session, test, params, vm, **driver_test_params)"
         % driver_test_name)'

I fix four parameters  we may use in the feature function. So It means if we have to transfer the four parameters when we define a  feature function.

The shortcoming:
we may need to define a useless parameter in some feature function.

The advantage：
We can use the same interface 'eval("win_driver_installer_test.%s_test( session, test, params, vm, **driver_test_params)"' .

If we want to avoid this issue, I  think of another way, we can use :
if hasattr(win_driver_installer_test,  driver_test_name):
   *args = XX
    getattr(win_driver_installer_test, driver_test_name)( *args)

but the codes will be a lot.

Reply:
> I will suggest to remove `session` and keep `vm`, then `session = vm.wait_for_login()`, as I suppose to open a session and then close it inside a function would be safer that pass it everywhere. Or at least `session` is not proper to be listed as a common parameter to `win_installer_test()`, we better not assume that all test need an opened session.

Thanks @vivianQizhu  That‘s a good idea, I will have a try.

Reply:
> I will suggest to remove `session` and keep `vm`, then `session = vm.wait_for_login()`, as I suppose to open a session and then close it inside a function would be safer that pass it everywhere. Or at least `session` is not proper to be listed as a common parameter to `win_installer_test()`, we better not assume that all test need an opened session.

Thanks @vivianQizhu  That's a good idea!  I will have a try.